subsys/dfu/boot/mcuboot.c

```diff
@@ -267,6 +383,25 @@ static int boot_magic_write(u8_t bank_id)
 	return rc;
 }
 
+static int boot_swap_type_write(u8_t bank_id, u8_t swap_type)
+{
+	const struct flash_area *fa;
+	u32_t offs;
+	int rc;
+
+	rc = flash_area_open(bank_id, &fa);
+	if (rc) {
+		return rc;
+	}
+
+	offs = SWAP_TYPE_OFFS(fa);
+
+	rc = boot_write_trailer_byte(fa, offs, swap_type);
+	flash_area_close(fa);
+
+	return rc;
+}
+
 static int boot_read_v1_header(u8_t area_id,
 			       struct mcuboot_v1_raw_header *v1_raw)
 {
```

Comment:
@mbolivar As I remember you have added this. So can you give us an idea what should be done with this? - for sure any changes to this should go in different PR.

Reply:
My idea at the time was that zephyr applications should be able to work with multiple versions of MCUboot. I.e. modern zephyr, old MCUboot, should continue to work forever, since MCUboot is considered non-upgradable.

Think of a device that needs to live for years.